mount.go

```diff
@@ -23,6 +23,8 @@ type mountFields struct {
 	subAppsProcessed sync.Once
 	// Prefix of app if it was mounted
 	mountPath string
+	// Parent app of the current app
```

Comment:
@ReneWerner87 do you think we should use full-mount path for mounting? I think this might be a bug related to MountPath

Reply:
MountPath can only see that of subApp. Currently, the app cannot know how the upper layer is mounted. So I introduce a reference to Parent here. Do you have any good suggestions?

My current thinking is that when PathRewrite in registerStatic can replace the mounted part of the path.

Reply:
@efectn @yinheli yes it should have something to do with the path
don't know if we necessarily need the complete parent reference for this

in the bug you said that the path is public/public, then we should start there

Reply:
> @efectn @yinheli yes it should have something to do with the path don't know if we necessarily need the complete parent reference for this
> 
> in the bug you said that the path is public/public, then we should start there

Instead, perhaps we can do it during the path setting https://github.com/gofiber/fiber/blob/main/mount.go#L52

Reply:
I have investigated this. However, here only the path of the subapp can be obtained. And `registerStatic` initializes the fileHandler when registering. So I chose the current implementation and added FullMountPath. I think that is easy to implement and has a lower cost.

Reply:
> I have investigated this. However, here only the path of the subapp can be obtained. And `registerStatic` initializes the fileHandler when registering. So I chose the current implementation and added FullMountPath. I think that is easy to implement and has a lower cost.

Ok i will check it tomorrow

Reply:
Appreciate

Reply:
Does the problem occur with v3 @yinheli ? I've just given a try with v3, and seems to work well:

```go
func TestStaticMount(t *testing.T) {
	t.Parallel()
	app := fiber.New()
	sub := fiber.New()
	sub2 := fiber.New()

	sub2.Get("/doe*", New("../../.github"))

	sub.Use("/sub2", sub2)
	app.Use("/sub", sub)

	resp, err := app.Test(httptest.NewRequest(fiber.MethodGet, "/sub/sub2/doe/release.yml", nil))
	require.NoError(t, err, "app.Test(req)")

	reqq, err := io.ReadAll(resp.Body)
	require.NoError(t, err)

	require.Equal(t, 200, resp.StatusCode, "Status code")
	require.Equal(t, uint32(1), app.HandlersCount())
	require.Equal(t, " ... ", string(reqq))
}
```

Reply:
I haven't investigated yet. I'm a bit busy during this period.